src/Chat/utils/__tests__/index.test.js

```diff
@@ -0,0 +1,31 @@
+import * as utils from '../index'
+
+describe('defer', () => {
+  const realSetImmediate = self.setImmediate
+
+  beforeEach(() => {
+    self.setImmediate = realSetImmediate
+  })
+
+  afterAll(() => {
+    self.setImmediate = realSetImmediate
+  })
+
+  test('should execute when setImmediate is defined', async () => {
+    const promise = new Promise(resolve => {
+      utils.defer(resolve)
+    })
+
+    await expect(promise).resolves.toBeUndefined()
+  })
+
+  test('should execute when setImmediate is undefined', async () => {
```

Comment:
Should this test be: _should execute `setTimeout` when `setIntermediate` is undefined_?

Reply:
Oh I see what's happening, the `promise` here is a spy.

[Jest](https://jestjs.io) has an excellent API for  [spies](https://jestjs.io/docs/en/mock-function-api), but this is great too.

Reply:
I thought about it as well, but I wanted to keep things simple.